### ISSUE TITLE
Add support for array in form data request body

### ISSUE THREAD

**Issue by yume-chan** [ferdikoomen/openapi-typescript-codegen#658]
**Describe the solution you'd like**

Practically, form data allows duplicate keys, and servers may parse them as an array (See https://stackoverflow.com/a/28434829).

One of our API requires such a parameter, but the generated requester doesn't support this usage.

From the above StackOverflow answer, different servers may behave differently, so maybe some extra investigations are required. For our backend written in PHP, the parameter name must end with `[]`.

**Comment by ferdikoomen** [comment]
@yume-chan Do you have an example (or snippet) of the OpenAPI spec? Then i can have a look. The way form data is handled is defined by OpenAPI: https://swagger.io/docs/specification/describing-request-body/

Specifically this section: "By default, arrays are serialized as array_name=value1&array_name=value2 and objects as prop1=value1&prop=value2, but you can use other serialization strategies as defined by the OpenAPI 3.0 Specification. The serialization strategy is specified in the encoding section"

There are a few encoding strategies: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md#encodingObject
They specify how the form data needs to be 'converted' to be accepted by the server.

However, this generator does not (yet) support this. Feel free to create a PR.

**Comment by yume-chan** [comment]
I'm still using OpenAPI 2.0 due to #257.

I checked OpenAPI 2.0 spec, the only mention to `array` type is [here](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#:~:text=number%22%2C%20%22integer%22%2C%20%22boolean%22%2C-,%22array%22,-or%20%22file%22.%20If).

I edit definition files using a program called [Stoplight Studio](https://stoplight.io/). It allows me selecting `array` as the type of a form data parameter. 

![image](https://user-images.githubusercontent.com/1330321/116388756-c1f66d00-a84e-11eb-94fc-08ac79f146df.png)

The generated code is like this:

```yml
  /user:
    post:
      summary: Create New User
      operationId: post-user
      responses:
        '200':
          description: User Created
          schema:
            $ref: '#/definitions/User'
        '400':
          description: Missing Required Information
          schema:
            type: 'null'
        '409':
          description: Email Already Taken
          schema:
            type: 'null'
      description: Create a new user.
      parameters:
        - in: formData
          type: array
          name: test
```

**Comment by yume-chan** [comment]
I found that it supports array in query string:

https://github.com/ferdikoomen/openapi-typescript-codegen/blob/830b317d0a0f6aaa97dba43818ce6d1ccc109be2/src/templates/core/functions/getQueryString.hbs#L1-L19

It should be similar for form data.

**Comment by ferdikoomen** [comment]
@yume-chan I published a new version 0.12.4, that should now support basic array types. Issue https://github.com/ferdikoomen/openapi-typescript-codegen/issues/772 is still valid

**Comment by yume-chan** [comment]
Thanks! I have verified it works for me.
